agent/drive_handler.go

```diff
@@ -146,3 +164,87 @@ func isStubDrive(d drive) bool {
 
 	return internal.IsStubDrive(f)
 }
+
+func (dh driveHandler) MountDrive(ctx context.Context, req *drivemount.MountDriveRequest) (*empty.Empty, error) {
+	logger := log.G(ctx).WithField("MountDriveRequest", req.String())
+	logger.Debug()
+
+	driveID := strings.TrimSpace(req.DriveID)
+	drive, ok := dh.GetDrive(driveID)
+	if !ok {
+		return nil, fmt.Errorf("Drive %q could not be found", driveID)
+	}
+	logger = logger.WithField("drive_path", drive.Path())
+
+	// Do a basic check that we won't be mounting over any important system directories
+	resolvedDest, err := evalAnySymlinks(req.DestinationPath)
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"failed to evaluate any symlinks in drive mount destination %q", req.DestinationPath)
+	}
+
+	for _, systemDir := range bannedSystemDirs {
+		if strings.HasPrefix(filepath.Clean(resolvedDest), filepath.Clean(systemDir)) {
+			return nil, errors.Errorf(
+				"drive mount destination %q resolves to path %q under banned system directory %q",
+				req.DestinationPath, resolvedDest, systemDir,
+			)
+		}
+	}
+
+	err = os.MkdirAll(req.DestinationPath, 0700)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create drive mount destination %q", req.DestinationPath)
+	}
+
+	const (
+		maxRetries = 100
+		retryDelay = 10 * time.Millisecond
+	)
+
+	for i := 0; i < maxRetries; i++ {
```

Comment:
In which cases drive might not be mountable?

Reply:
This can happen when we've patched a stub drive but the OS has not yet realized a change was made, which is rare but did happen in practice occasionally. We needed these retries in the previous implementation too, I just migrated the code as part of this change.

Reply:
nit: I think this is useful context and it worth to leave comments explaining the motivation behind retries.

Reply:
Yeah good call, added a comment w/ link to original issue